app/models/account/states.rb

```diff
@@ -143,12 +143,6 @@ def will_be_deleted?
       scheduled_for_deletion? || (buyer? && provider_account.try(:should_be_deleted?))
     end
 
-    def ready_to_be_suspended?
-      tenant? || marked_for_suspension
-    end
-
-    attr_accessor :marked_for_suspension
```

Comment:
I'm thinking that `marked_for_suspension` was kind of a asfety check for the automation process, not to suspend unmarked accounts.

I wonder if it makes sense to keep it. Just in the UI call to mark for suspension before actually suspending in case that is needed.

... in case what I think makes any sense to begin with

Reply:
I removed the `marked_for_suspension` attribute at the beginning because it was interfering with the UI: you couldn't see the suspend button unless the buyer was already marked for suspension. I did some refactoring and now it works as expected and keeps `marked_for_suspension`.

I force-pushed, but you can see the changes here: https://github.com/3scale/porta/compare/5acad5442edd450a3b20f35f73125d26d681fcb1..4f7efa4ed9fdfde74a159e7687df74250335e594